discovery-provider/src/queries/search.py

```diff
@@ -61,193 +61,222 @@ def search_tags():
     if not user_tag_count:
         user_tag_count = "2"
 
+    kind = request.args.get("kind", type=str)
+    if kind:
+        validSearchKinds = [SearchKind.all, SearchKind.tracks, SearchKind.users]
+        error = False
+        try:
+            searchKind = SearchKind[kind]
+        except Exception as e:
+            error = True
+        if (searchKind not in validSearchKinds or error):
+            raise exceptions.ArgumentError(
+                "Invalid value for parameter 'kind' must be in %s" %
+                    [k.name for k in validSearchKinds]
+            )
+    else:
+        searchKind = SearchKind.all
+
+    results = {}
+
     (limit, offset) = get_pagination_vars()
     like_tags_str = str.format('%{}%', search_str)
     db = get_db()
     with db.scoped_session() as session:
-        track_res = sqlalchemy.text(
-            f"""
-            select distinct(track_id)
-            from
-            (
-                select
-                    strip(to_tsvector(tracks.tags)) as tagstrip,
-                    track_id
+        if (searchKind in [SearchKind.all, SearchKind.tracks]):
+            track_res = sqlalchemy.text(
+                f"""
+                select distinct(track_id)
```

Comment:
NIT - indentation

Reply:
what's wrong with the indentation here?

Reply:
oh i think i see what you meant, cleaned up a bit